examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -169,56 +169,15 @@
 #
 # Gradient Boosting Regression with decision trees is often flexible enough to
 # efficiently handle heteorogenous tabular data with a mix of categorical and
-# numerical features as long as the number of samples is large enough.
-#
-# Here, we do minimal ordinal encoding for the categorical variables and then
-# let the model know that it should treat those as categorical variables by
-# using a dedicated tree splitting rule. Since we use an ordinal encoder, we
-# pass the list of categorical values explicitly to use a logical order when
-# encoding the categories as integers instead of the lexicographical order.
-# This also has the added benefit of preventing any issue with unknown
-# categories when using cross-validation.
-#
-# The numerical variables need no preprocessing and, for the sake of simplicity,
-# we only try the default hyper-parameters for this model:
+# numerical features as long as the number of samples is large enough. We set
+# `categorical_features="by_dtype"` such that features with categorical dtype
+# are considered categorical features.
 from sklearn.pipeline import make_pipeline
-from sklearn.preprocessing import OrdinalEncoder
 from sklearn.compose import ColumnTransformer
 from sklearn.ensemble import HistGradientBoostingRegressor
 from sklearn.model_selection import cross_validate
 
-
-categorical_columns = [
-    "weather",
-    "season",
-    "holiday",
-    "workingday",
-]
```

Comment:
Could we select them automatically from pandas and print them out?

Reply:
Included this suggestion in [`060200f` (#26411)](https://github.com/scikit-learn/scikit-learn/pull/26411/commits/060200f3f9da4c895d36d8888a4f16aab583206c).